addons/web/controllers/export.py

```diff
@@ -328,42 +328,99 @@ def get_fields(self, model, prefix='', parent_name='',
+                for definition in record[definition_record_field]:
+                    # definition = {
+                    #     'name': 'aa34746a6851ee4e',
+                    #     'string': 'Partner',
+                    #     'type': 'many2one',
+                    #     'comodel': 'test_new_api.partner',
+                    #     'default': [1337, 'Bob'],
+                    # }
+                    if (
+                        definition['type'] == 'separator' or
+                        (
+                            definition['type'] in ('many2one', 'many2many')
+                            and definition['comodel'] not in Model.env
+                        )
+                    ):
+                        continue
+                    id_field = f"{fname}.{definition['name']}"
+                    property_fields[id_field] = {
+                        'type': definition['type'],
+                        'string': Model.env._(
+                            "%(property_string)s (%(parent_name)s)",
+                            property_string=definition['string'], parent_name=record.display_name,
+                        ),
+                        'default_export_compatible': field['default_export_compatible'],
+                    }
+                    if definition['type'] in ('many2one', 'many2many'):
+                        property_fields[id_field]['relation'] = definition['comodel']
+
+        exportable_fields.update(property_fields)
+
+        fields_sequence = sorted(exportable_fields.items(), key=lambda field: field[1]['string'].lower())
+
+        result = []
+        for field_name, field in fields_sequence:
             ident = prefix + ('/' if prefix else '') + field_name
             val = ident
             if field_name == 'name' and import_compat and parent_field_type in ['many2one', 'many2many']:
                 # Add name field when expand m2o and m2m fields in import-compatible mode
                 val = prefix
             name = parent_name + (parent_name and '/' or '') + field['string']
-            record = {'id': ident, 'string': name,
-                      'value': val, 'children': False,
-                      'field_type': field.get('type'),
-                      'required': field.get('required'),
-                      'relation_field': field.get('relation_field'),
-                      'default_export': import_compat and field.get('default_export_compatible')}
-            records.append(record)
-
+            field_dict = {
+                'id': ident,
+                'string': name,
+                'value': val,
+                'children': False,
+                'field_type': field.get('type'),
+                'required': field.get('required'),
+                'relation_field': field.get('relation_field'),
+                'default_export': import_compat and field.get('default_export_compatible')
+            }
             if len(ident.split('/')) < 3 and 'relation' in field:
```

Comment:
The `< 3` should have a comment with an example.

Reply:
IMO, I think you are right (it checks `field.relational`) but I am not 100% sure about the implications of this. Then I rather avoid changing it, if it is ok for you ?

Reply:
Ok, you may leave a TODO if you want.